packages/curve-ui-kit/src/hooks/useDebounce.ts

```diff
@@ -70,4 +70,25 @@ function useDebounce<T>(initialValue: T, debounceMs: number, callback: (value: T
   return [value, setDebouncedValue] as const
 }
 
-export default useDebounce
+const SearchDebounceMs = 166 // 10 frames at 60fps
+
+/**
+ * A hook that debounces a search value and calls a callback when the debounce period has elapsed.
+ */
+export function useSearchDebounce<T>(defaultValue: T, callback: (value: T) => void, debounceMs = SearchDebounceMs) {
```

Comment:
Thinking about this, maybe we should rename it to 'useUniqueDebounce' or something? It's not that related to searching anymore (well, until the return of `if (typeof value === 'string')`.